### ISSUE TITLE
run_analysis.R

### ISSUE THREAD

**Issue by seba7777777** [seba7777777/Getting-and-Cleaning-data#3]
# /_Run_analisys.R_/
# /*Getting and Cleaning Data coursea */
# /_Sebastian Gierz_/
# /_Data source_/
# /*Human Activity Recognition Using Smartphones Dataset Version 1.0  */
# /_Downloaded :_/
# /_C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean_/
# -------------------------------------------------------------------------------------------------*/
# import and prepare files*/
# -------------------------------------------------------------------------------------------------*/

  th_train <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\train\subject_train.txt")
  th_test <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\test\subject_test.txt")
  train_dataset <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\train\X_train.txt") 
  test_dataset <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\test\X_test.txt") 
  Y_train <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\train\Y_train.txt") 
  Y_test <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\test\Y_test.txt") 

  # ids for th files 
  names(th_train) <- "thID"
  names(th_test) <- "thID"

  # connect column names from file features.txt
  featureNames <- read.table("C:\Users\sebastian.gierz\Desktop\WORK\R\Hopkins_univ\clean\getdata-projectfiles-UCI HAR Dataset\UCI HAR Dataset\features.txt")
  names(train_dataset) <- featureNames$V2
  names(test_dataset) <- featureNames$V2

  # Column name for Y files
  names(Y_train) <- "activity"
  names(Y_test) <- "activity"
# /_-------------------------------------------------------------------------------------------------_/
# /_ad1.Merges the training and the test sets to create one data set._/
# /_-------------------------------------------------------------------------------------------------_/

  ds_train <- cbind(th_train, Y_train, train_dataset)
  ds_test <- cbind(th_test, Y_test, test_dataset)
  FULL_dataset <- rbind(ds_train, ds_test)

  # just for control ;)
  #head(FULL_dataset)

  #/_-------------------------------------------------------------------------------------------------_/  
  #/_ad2.Extracts only the measurements on the mean and standard deviation for each measurement. 
  #/_-------------------------------------------------------------------------------------------------*/

  # select columns with mean() and std()
  columns_MS <- grepl("mean\(\)", names(FULL_dataset)) |
    grepl("std\(\)", names(FULL_dataset))

  # and save thID and activity columns
  columns_MS[1:2] <- TRUE

  # and we don't need any more other columns
  FULL_dataset <- FULL_dataset[, columns_MS]

  #/_-------------------------------------------------------------------------------------------------_/  
  #/_ad3.Uses descriptive activity names to name the activities in the data set. 
  #/_-------------------------------------------------------------------------------------------------_/
  #/_-------------------------------------------------------------------------------------------------_/  
  #/_ad4.Appropriately labels the data set with descriptive variable names. 
  #/_-------------------------------------------------------------------------------------------------_/

  # change column with activity to factor with correct labels
  FULL_dataset$activity <- factor(FULL_dataset$activity, labels=c("Walking", "Walking Upstairs", "Walking Downstairs", "Sitting", "Standing", "Laying"))

  #/_-------------------------------------------------------------------------------------------------_/  
  #/_ad5.From the data set in step 4, creates a second, independent tidy data set with the average of each variable for each activity and each subject. 
  #/_-------------------------------------------------------------------------------------------------*/

  # to use function "melt" in this case, i will need package reshape2
  library(reshape2)
  # create base and tidy data set
  dataset_m <- melt(FULL_dataset, id=c("thID","activity"))
  tidy_ds <- dcast(dataset_m, thID+activity ~ variable, mean)

  #head(tidy_ds)

  #write results as tidy data

  write.table(tidy_ds, file = "./results_as_thidy_file.txt", row.names=FALSE)
